### PR TITLE
grdinfo: Fix pad for external wrappers and update grid header for -L option

### PR DESCRIPTION
**Pull request by seisman**
Closes https://github.com/GenericMappingTools/gmt/issues/8525.

### CODE REVIEW

File: src/grdinfo.c [hunk 1]
```diff
@@ -689,8 +689,6 @@ EXTERN_MSC int GMT_grdinfo (void *V_API, int mode, void *args) {
 
 	Out = gmt_new_record (GMT, (n_cols) ? out : NULL, (cmode) == GMT_COL_FIX ? record : NULL);	/* the two items */
 
-	gmt_set_pad (API->GMT, 0);	/* Not read pads to simplify operations */
```
Comment (seisman) [inline src/grdinfo.c:692]: @joa-quim Do you see any side effects removing this line?

**Comment by joa-quim** [comment]
_simplify operations_ does not seems a reason good enough. Specially when it has buggish side effects, so I would say we should remove it and see if there are any noticeable side effects. 

The pad is big annoyance for externals transfering grids in memory because it implies making a full copy of the array and for many operations, grdinfo for example, the pad is not necessary at all. In Julia I try to distinguish uses where pad is needed (projections, gradients, etc...) from others where it doesn't and in those case I set the pad to zero.

**Review by joa-quim — APPROVED** [review]

**Comment by seisman** [comment]
No new failures and the original bug is fixed. Merging.

**Comment by joa-quim** [comment]
Well, a little bit by reasoning + trial-error. Making a projected image needs pad but a linear one (-JX) does not. Shading needs pad because of the gradient. Also, when a module needs the pad and grid/image do not have it, an error is issued.

**Comment by seisman** [comment]
Good to know. Will see how it works in PyGMT.

**Pull request merged**